### PR TITLE
enable xpu group norm channels last support

### PR DESCRIPTION
**Pull request by KevinYuk**
XPU would support channels last format for group norm operator, however, Pytorch converts all input tensor to contiguous format, which includes channels last tensor. Need Pytorch pass down this memory format hint to us.

### CODE REVIEW

File: tools/autograd/derivatives.yaml [hunk 1]
```diff
@@ -1160,7 +1160,7 @@
...
-  input, weight, bias: "GradMode::is_enabled() || grads[1].defined() || grads[2].defined() ? infinitely_differentiable_native_group_norm_backward(grads[0], grads[1], grads[2], input, result1, result2, weight, N, C, HxW, group, eps, grad_input_mask) : (grads[0].defined() ? native_group_norm_backward_symint(grads[0].is_contiguous() ? grads[0] : grads[0].contiguous(), input.is_contiguous() ? input : input.contiguous(), result1, result2, weight, N, C, HxW, group, grad_input_mask) : std::tuple<Tensor, Tensor, Tensor>())"
+  input, weight, bias: "GradMode::is_enabled() || grads[1].defined() || grads[2].defined() ? infinitely_differentiable_native_group_norm_backward(grads[0], grads[1], grads[2], input, result1, result2, weight, N, C, HxW, group, eps, grad_input_mask) : (grads[0].defined() ? native_group_norm_backward_symint(grads[0].device().is_xpu() ? grads[0] : grads[0].is_contiguous() ? grads[0] : grads[0].contiguous(), input.device().is_xpu() ? input : input.is_contiguous() ? input : input.contiguous(), result1, result2, weight, N, C, HxW, group, grad_input_mask) : std::tuple<Tensor, Tensor, Tensor>())"
```
Comment (albanD) [inline tools/autograd/derivatives.yaml:1163]: wow this is hard to read :p 
Could you replace `grads[0].is_contiguous() ? grads[0] : grads[0].contiguous()` with `grads[0].contiguous()` there please as they are equivalent.

File: aten/src/ATen/native/group_norm.cpp [hunk 2]
```diff
@@ -181,7 +181,7 @@ Tensor group_norm(
 
   const Tensor kEmpty;
   auto memory_format = input.suggest_memory_format();
-  const auto& X = input.device().is_cpu() ?
+  const auto& X = input.device().is_cpu() || input.device().is_xpu() ?
```
Comment (albanD) [inline aten/src/ATen/native/group_norm.cpp:184]: This is not following what cuda does here? That is expected?

File: tools/autograd/derivatives.yaml (hunk 1 above)
Comment (KevinYuk) [reply tools/autograd/derivatives.yaml]: Yes. Thanks for your suggestion.

File: aten/src/ATen/native/group_norm.cpp (hunk 2 above)
Comment (KevinYuk) [reply aten/src/ATen/native/group_norm.cpp:184]: hi @albanD cuda doesn't support channels last here. Before only cpu is supported, and now we want to support it for xpu.

**Review by albanD — APPROVED** [review]
Thanks for the cleanup!

**Comment by KevinYuk** [comment]
@soulitzer Please help. Thanks a lot.

**Comment by albanD** [comment]
@KevinYuk no need for an accept from all the reviewers (unless otherwise specified by the person that does the first accept of course).
This can be landed!

**Pull request closed without merging**